cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/artifact/ArtifactRepository.java

```diff
@@ -186,12 +186,40 @@ public ArtifactDetail getArtifact(Id.Artifact artifactId) throws IOException, Ar
    *
    * @param artifactId the id of the artifact to inspect and store
    * @param artifactFile the artifact to inspect and store
+   * @return detail about the newly added artifact
+   * @throws IOException if there was an exception reading from the artifact store
+   * @throws WriteConflictException if there was a write conflict writing to the ArtifactStore
+   * @throws ArtifactAlreadyExistsException if the artifact already exists
+   * @throws InvalidArtifactException if the artifact is invalid. For example, if it is not a zip file,
+   *                                  or the application class given is not an Application.
+   */
+  public ArtifactDetail addArtifact(Id.Artifact artifactId, File artifactFile)
```

Comment:
functionally yes, but this method doesn't throw an exception about parent artifacts not found whereas that one does.